pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -1,231 +1,284 @@
+    el.classList.remove("invalid");
+    el.classList.add("valid");
+  };
+
+  const checkFail = (el) => {
+    el.classList.remove("valid");
+    el.classList.add("invalid");
   };
 };
 ```
 
-User actions can trigger [mutations within the `amp-script` component ](../../../documentation/components/reference/amp-script.md?format=websites#mutations-and-user-actions), allowing `amp-script` to register [event listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventListener).
+Let's make those `valid` and `invalid` classes actually turn text green or red. Go back to `index.html`, and add these two rules to the `<style amp-custom>` tag:
 
-Our function will listen for two events, [`keyup`](https://developer.mozilla.org/en-US/docs/Web/API/Document/keyup_event) for when a user types into the input box, and [`change`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/change_event) in case our user pastes their password.
+```css
+      li.valid {
+        color: #2d7b1f;
+      } 
 
-```js
-function initCheckPassword(element) {
- const checkPassword = () => {
-    passwordChecks.forEach((item) => {
-      let passed = item.test(element.value);
-       // passed/fail logic
-    });
-    // is called when user types in input
-  element.addEventListener("keyup", checkPassword);
-  // is called if user pastes into input
-  element.addEventListener("change", checkPassword);
-  };
-};
+      li.invalid {
+        color:#c11136;
+      }
 ```
 
-Inside the `checkPassword` function, we’ll toggle a class that changes our text to green if the check is passed.
+Now we're ready to add the logic that checks the contents of the password `<input>` against our rules. Add a new function called `checkPassword()` to `initCheckPassword()`, right before the closing brace:
 
 ```js
-function initCheckPassword(element) {
   const checkPassword = () => {
-    passwordChecks.forEach((item) => {
-      let passed = item.test(element.value);
-      // captures element
-      let checkText = document.getElementById(item.elementId)
-       // passed/fail logic
-       checkText.classList.toggle('pass', passed)
-    });
-  }
-    // is called when user types in input
-    element.addEventListener("keyup", checkPassword);
-    // is called if user pastes into input
-    element.addEventListener("change", checkPassword);
-};
+    const password = element.value;
+    let failed = false;
+
+    for (const check in checkRegexes) {
+      let li = document.getElementById(check);
+
+      if (password.match(checkRegexes[check])) {
+        checkPass(li);
+      } else {
+        checkFail(li);
+        failed = true;
+      }
+    }
+
+    if (!failed) {
+      submitButton.removeAttribute("disabled");
+    }
+  };
 ```
 
-For this to work, we will need to add a `check` class to all the items and define both `check` and `pass` within the `<style amp-custom>` tag in the page head.
+This function does the following:
 
+* Grabs the contents of the password `<input>`.
```

Comment:
suggestion: use an enumeration instead of bullets for the top level listing

Reply:
I like it